Lib/Protocols/IdCompressorZLib.pas

```diff
@@ -172,7 +172,11 @@ procedure TIdCompressorZLib.InternalDecompressStream(
     repeat
       LZstream.next_out := @outBuffer[0];
       LZstream.avail_out := bufferSize;
-      DCheck(inflate(LZstream,Z_NO_FLUSH));
+      zresult := inflate(LZstream, Z_NO_FLUSH);
+      if zresult <> Z_BUF_ERROR then
+      begin
+        DCheck(zresult);
+      end;
       outSize := bufferSize - LZstream.avail_out;
       AOutStream.Write(outBuffer, outSize);
     until (LZstream.avail_in = 0) and (LZstream.avail_out > 0);

```

Comment:
Is there ever a situation where ignoring `Z_BUF_ERROR` may lead to an endless loop here?

Reply:
This has never happened in my work